pydicom/dataset.py

```diff
@@ -57,6 +57,57 @@ class PropertyError(Exception):
     pass
 
 
+class PrivateBlock(object):
+    """Helper class for a private block in the dataset.
+    (See PS3.5, Section 7.8.1 - Private Data Element Tags)
+
+    Attributes
+    ----------
+    group : 32 bit int
+        The private group where the private block is located.
+    private_creator : str
+        The private creator string related to the block.
+    dataset : Dataset
+        The parent dataset.
+    block_start : 32 bit int
+        The start element of the private block.
+        Note that the 2 low order hex digits of the element are always 0.
+    """
+
+    def __init__(self, key, dataset, private_creator_element):
+        """Initializes an object corresponding to a private tag block.
+
+        Parameters
+        ----------
+        key : tuple (int, str)
+            The private group and private creator. The group must be an odd
+            number.
+        dataset : Dataset
+            The parent dataset.
+        private_creator_element : 32 bit int
+            The element of the private creator tag.
+        """
+        self.group = key[0]
+        self.private_creator = key[1]
+        self.dataset = dataset
+        self.block_start = private_creator_element << 8
+
+    def real_tag(self, tag):
+        """Return the real private tag ID for the given tag.
+        Note that the 2 high order hex digits of the tag element are ignored
+        and replaced by the digits defined by the block start.
+        """
+        tag = Tag(tag)
+        return Tag(tag.group, (tag.element & 0xff) | self.block_start)
+
+    def add_new(self, tag, VR, value):
+        """Adds the given tag to the parent dataset - see
```

Comment:
I forgot to change the `tag` parameter to the 16 bit element - will change this later

Reply:
Ok, done.